crates/release_plz/src/args/shell_completions.rs

```diff
@@ -0,0 +1,23 @@
+use anyhow::anyhow;
+use clap::IntoApp;
+use clap_complete::Shell;
+use std::str::FromStr;
+
+#[derive(clap::Parser, Debug)]
+pub struct ShellCompletions {
+    /// Shell option
+    #[clap(default_value = "bash")]
+    shell: String,
+}
+
+impl ShellCompletions {
+    pub fn print(&self) -> anyhow::Result<()> {
+        clap_complete::generate(
+            Shell::from_str(&self.shell).map_err(|e| anyhow!("Shell {e}"))?,
+            &mut super::CliArgs::command(),
+            "release_plz",
+            &mut std::io::stdout(),
+        );
+        Ok(())
```

Comment:
```suggestion
    shell: Shell,
}

impl ShellCompletions {
    pub fn print(&self) {
        clap_complete::generate(
            self.shell,
            &mut super::CliArgs::command(),
            "release-plz",
            &mut std::io::stdout(),
        );
```

## Change 1: fix binary name

The name of the binary is `release-plz`, not `release_plz`. I fixed that :)

## Change 2: use the `Shell` type in the ShellCompletions struct

By using the `Shell` type directly in the `ShellCompletions` struct you let clap do the input validation for you.
As you can see the code is also much cleaner because you don't have to handle the error of an invalid shell.

Let's compare the error messages for the invalid input `dsadsa`:

### Error messages before this change

![image](https://user-images.githubusercontent.com/11428655/173193626-3d5a26c6-b3d7-4e74-a405-c6f430a6b78c.png)

### Error message after this change
![image](https://user-images.githubusercontent.com/11428655/173193759-7e467d11-1c62-4a8a-ac9c-64261a921319.png)

This one looks nicer to me!
What do you think? :)

Reply:
Awesome suggestion! I haven't even thought we could use something other than a String for the ShellCompletions struct.